contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -248,15 +262,14 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
         override
         public
     {
-        uint40 nextQueueIndex = getNextQueueIndex();
-        uint40 queueLength = _getQueueLength();
-        _numQueuedTransactions = Math.min(_numQueuedTransactions, queueLength - nextQueueIndex);
+        _numQueuedTransactions = Math.min(_numQueuedTransactions, getNumPendingQueueElements());
```

Comment:
Wouldn't this be a sign that there is a reorg in the honest sequencer case?  Could even throw if _numqueued exceeds numpending.

Reply:
Ah you’re right but this is used a little differently. So when the sequencer goes down and we need to clear the queue, the batch submitter just blindly calls this with a massive number to make sure that the queue is for sure emptied.

Reply:
And then this is also useful for normal nodes if they ever want to clear the queue without any race conditions

Reply:
great, ship it then!